File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
    if len(nums) < 1 {
        return 0
    }

    numsMap := map[int]bool{}
    for _, num := range nums {
        numsMap[num] = true
    }

    maxLen := 0
    currLen := 0
    for num := range numsMap {
        if _, ok := numsMap[num - 1]; !ok {
            currNum := num
            currLen = 1
            for numsMap[currNum + 1] {
                currNum += 1
                currLen += 1
            }

            if maxLen < currLen {
                maxLen = currLen
            }
        }
    }

    if maxLen < currLen {
        maxLen = currLen
    }

    return maxLen;
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) < 1:
            return 0

        num_map = {num: True for num in nums}

        max_length = 0
        for num in num_map:
            if num - 1 not in num_map:
                current_length = 1
                current_num = num
                while current_num + 1 in num_map:
                    current_length += 1
                    current_num += 1

                max_length = max(max_length, current_length)
        
        return max_length
